semantics/src/traversal/expressions.rs

```diff
@@ -256,6 +256,34 @@ fn expr_bin_operation(
     unreachable!()
 }
 
+fn expr_call(
+    scope: Shared<FunctionScope>,
+    exp: &Spanned<fe::Expr>,
+) -> Result<ExpressionAttributes, SemanticError> {
+    if let fe::Expr::Call { args: _, func } = &exp.node {
+        if let fe::Expr::Attribute { value: _, attr } = &func.node {
+            let contract_scope = &scope.borrow().parent;
+            let called_func = contract_scope.borrow().def(attr.node.to_string());
+            return match called_func {
+                Some(ContractDef::Function {
+                    params: _,
+                    returns: Some(_),
+                }) => Ok(ExpressionAttributes {
+                    typ: Type::Base(Base::U256),
+                    location: Location::Value,
+                }),
+                Some(ContractDef::Function {
+                    params: _,
+                    returns: None,
+                }) => Err(SemanticError::TypeError),
```

Comment:
I'm not exactly sure if `TypeError` is the right choice here. If we had a `void` type, I'd say it is but since Fe doesn't have a concept of returning `void` if it doesn't explicitly returns something, maybe a different error type needs to be introduced. What do you think @g-r-a-n-t ?

Reply:
`NotAnExpression` might be more fitting. 

In general, feel free to create new error types on a whim. I'm thinking it would be better to perform a refactor of errors once the library is built out more fully than to try to get each error right in every PR.

Eventually it would be nice to print error like Rust does:

```
return self.foo()
       ----------
           ^
           error: not an expression

hint: consider giving `foo` a return value.
```

So we'll have to designate some time to focus solely on improving errors at some point.

Reply:
Alright cool! Will update with `NotAnExpression` :+1: